lexerState.go

```diff
@@ -57,6 +58,7 @@ var validLexerStates = []lexerState{
 			COMPARATOR,
 			LOGICALOP,
 			CLAUSE_CLOSE,
+			TERNARY,
```

Comment:
You probably don't want it to be valid to transition between numeric and ternary - the ternary operator only operates on booleans (which i think is a good choice), so it's best if only the `BOOLEAN` and `VARIABLE` lexer states were allowed to transition to ternary.